edx_rest_api_client/auth.py

```diff
@@ -45,7 +45,7 @@ def __call__(self, r):
             data['tracking_context'] = self.tracking_context
 
         set_custom_metric('deprecated_jwt_signing', 'JwtAuth')
-        r.headers['Authorization'] = 'JWT {jwt}'.format(jwt=jwt.encode(data, self.signing_key))
+        r.headers['Authorization'] = 'JWT {jwt}'.format(jwt=jwt.encode(data, self.signing_key).decode("UTF-8"))
```

Comment:
Hello @morenol.  I just used the above metric `deprecated_jwt_signing` to find usage of this, wondering why we never ran into any errors.  It seems that we aren't using this code, so it is probably better to remove this than fix it.

We don't have a metric in `__init__`, which we could add before removing the whole class.

Thoughts?

Reply:
As far as I know we are using this in ecommerce-worker and we have not get into this issue because ecommerce-worker is running with python2, in fact ecommerce-worker runs the 1.8.2 version of this repo that also has this problem.

 I don;t know if there are other services/libraries that are using this.

Reply:
I think that most of the services have migrated to SuppliedJwtAuth because they use the 
get_and_cache_oauth_access_token function to generate the JWT token, 

I think that currently  that is not possible for ecommerce-worker